src/main/kotlin/org/opensearch/commons/alerting/model/UnifiedAlert.kt

```diff
@@ -0,0 +1,208 @@
+package org.opensearch.commons.alerting.model
+
+import org.opensearch.common.lucene.uid.Versions
+import org.opensearch.commons.alerting.util.IndexUtils.Companion.NO_SCHEMA_VERSION
+import org.opensearch.commons.alerting.util.instant
+import org.opensearch.commons.alerting.util.optionalTimeField
+import org.opensearch.commons.alerting.util.optionalUserField
+import org.opensearch.commons.authuser.User
+import org.opensearch.core.common.io.stream.StreamInput
+import org.opensearch.core.common.io.stream.StreamOutput
+import org.opensearch.core.common.io.stream.Writeable
+import org.opensearch.core.xcontent.ToXContent
+import org.opensearch.core.xcontent.XContentBuilder
+import org.opensearch.core.xcontent.XContentParser
+import org.opensearch.core.xcontent.XContentParserUtils
+import java.io.IOException
+import java.time.Instant
+
+/** CorrelationAlert and Alert can extend the UnifiedAlert class to inherit the common fields and behavior
+ * of UnifiedAlert class.
+ */
+open class UnifiedAlert(
```

Comment:
why is Alert class not extending this class?

Reply:
we can do that at a later stage, when we will start Push Notifications and  whenever we onboard MonitorBased Alerts onto it. Right now focus for this PR is just Correlation based alerts